scripts/exps/kani-run-on-repos.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+
+DOCUMENTATION=\
+'kani-run-on-repos.sh -- script to clone and compile multiple remote git repositories with Kani.
+
+USAGE:
+./scripts/kani-run-on-repos.sh path/to/url-list
+
+Download the top 100 crates and runs kani on them. Prints out the
+errors and warning when done. Xargs is required for this script to
+work.
+
+url-list: A list of URLs to run Kani on. One per line.
+
+ENV:
+- PRINT_STDOUT=1 forces this script to search for warning in
+  STDOUT in addition to STDERR
+
+EDITING:
+- To adjust the git clone or kani args, modify the function
+  `clone_and_run_kani`.
+- To adjust the errors this script searches for, edit the function
+  `print_errors_for_each_repo_result`
+
+Copyright Kani Contributors
+SPDX-License-Identifier: Apache-2.0 OR MIT'
+
+export SELF_SCRIPT=$0
+export SELF_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+NPROC=$(nproc 2> /dev/null || sysctl -n hw.ncpu 2> /dev/null || echo 4)  # Linux or Mac or hard-coded default of 4
+export WORK_DIRECTORY_PREFIX="$SELF_DIR/../target/remote-repos"
+
+
+export STDOUT_SUFFIX='stdout.cargo-kani'
+export STDERR_SUFFIX='stderr.cargo-kani'
+export EXIT_CODE_SUFFIX='exit-code.cargo-kani'
+# worker function that clones target repos and runs kani over
+# them. This functions is called in parallel by
+# parallel_clone_and_run, and should not be run explicitly
+function clone_and_run_kani {
+    WORK_NUMBER_ID=$(echo $1 | awk -F ',' '{ print $1}')
+    REPOSITORY_URL=$(echo $1 | awk -F ',' '{ print $2}')
+    REPO_DIRECTORY="$WORK_DIRECTORY_PREFIX/$WORK_NUMBER_ID"
+    echo "work# $WORK_NUMBER_ID -- $REPOSITORY_URL"
+
+    # clone or update repository
+    (git clone $REPOSITORY_URL $REPO_DIRECTORY 2> /dev/null || git -C $REPO_DIRECTORY pull)
+
+    # run cargo kani compile on repo. save results to file.
+    PATH=$PATH:$SELF_DIR
+    (cd $REPO_DIRECTORY; nice -n15 cargo kani --only-codegen) \
+         1> $REPO_DIRECTORY/$STDOUT_SUFFIX \
+         2> $REPO_DIRECTORY/$STDERR_SUFFIX
+    echo $? > $REPO_DIRECTORY/$EXIT_CODE_SUFFIX
+}
+export -f clone_and_run_kani
+
+OVERALL_EXIT_CODE='0'
+TARGET_ERROR_REGEX='warning:\sFound\sthe\sfollowing\sunsupported\sconstructs:\|WARN'
+# printing function that greps the error logs and exit code.
+function print_errors_for_each_repo_result {
+    DIRECTORY=$1
+    IS_FAIL='0'
+
+    error_code="$(cat $DIRECTORY/$EXIT_CODE_SUFFIX)"
+    if [ "$error_code" != "0" ]; then
```

Comment:
Can you please print where the logs are?

Reply:
Implemented with f2e5d10c9a250bfb9c3805671108031172295b9f.